(charts): display more space on top of charts for better readability

defymecobra opened the pull request:
Added optimal Y-axis calculation in `sharedUtils.ts`:
- Implemented `calculateOptimalInterval()` to determine clean interval values (1, 2, 3, 5, 10, etc.) based on data range
- Implemented `calculateOptimalMax()` to round the Y-axis maximum up to the next optimal division
- This ensures consistent spacing between all grid lines and adds visual breathing room at the top

Now they look like this:

<img width="2279" height="1207" alt="image" src="https://github.com/user-attachments/assets/e5e90a42-6a38-4311-bdf0-61ba77081c77" />
<img width="2289" height="606" alt="image" src="https://github.com/user-attachments/assets/fb835213-8e2c-4f4a-9ec6-daf274c688ef" />
<img width="2292" height="1205" alt="image" src="https://github.com/user-attachments/assets/54222012-4e66-4e34-9ba7-86c50072601f" />


### ALSO

Added proper maximum value calculation for stacked bar charts in `BarChartWidget.tsx`:
- Implemented `actualMaxValue` computation that correctly sums stacked values
- Previous implementation only considered individual series maximums, not the total stack height
- Now passes the correct maximum to `generateYAxis()`, ensuring proper scaling for stacked bars

defymecobra commented:
@rorychatt pls look at changes in `BarChartWidget.tsx`. With my new logic in `sharedUtils`, Area and Line charts work correctrly, but bar isn't, cause the logic of maxValue in it was broken until I've made my changes

rorychatt commented:
@defymecobra I think solution should be way simpler and most likely is some config we need to update

https://echarts.apache.org/examples/en/editor.html?code=PYBwLglsB2AEC8sDeAoWsDGwA2wBOAXLANoBEAxAMyUCcAHACIBCpAugDRqxjA6QhFU6dGDwQA5uICmhWKQCGADwgBnUp2GwlqgArAI0MDMFdNYAJ4gpRUioAW8gCbAA7qVOwAvl08bY4sUdBXy5FAEFlFSJiIWELKxsMeSNxfHN1D0dk-WjSAFkYdTkAFQBXKSLSAHUpR0riu1LKgDExSoBlZI7S6DYfDi5zCNVo2JFLazkAN3lscozNAHpF2AAjYB6svHMAcXkBEgAGdlhD1n6_FRkIKSiSMdhoeQBbSdJAF7dAUuNAfujAO9SFuITGyreR4AHoEF4KoQRxgOw2ABshwApODYFkwDkSABGY6wACsACYToTDnjqAAWE4U3Enah4wmk87oTznTxAA&_source=echarts-doc-preview

See this -> it renders properly in their examples

rorychatt reviewed (changes requested):
there has to be another solution that was made by the library

IDK maybe this is not it, you have to try
https://echarts.apache.org/en/option.html#yAxis.boundaryGap

defymecobra commented:
Deleted the previous complex solution and implemented a simpler solution using conditional max property:
- When largeSpread is false, the max property is not added to the axis configuration at all
- ECharts' native auto-scaling algorithm kicks in, automatically calculating optimal Y-axis range
- This provides the desired "more spacing" at the top of charts without custom logic
- ECharts ensures consistent intervals and rounds to nice numbers (100, 200, 300, etc.)
- Works correctly for all chart types including stacked bar charts

This approach leverages ECharts' built-in intelligence rather than reimplementing it, resulting in cleaner, more maintainable code.

Results (all works correclty):

<img width="1116" height="543" alt="image" src="https://github.com/user-attachments/assets/ee6d8181-adee-4e1b-b2b9-c27adef9cd0e" />
<img width="1103" height="538" alt="image" src="https://github.com/user-attachments/assets/4dfc21c0-cccf-4bad-bea4-463a5b4bc40a" />
<img width="1110" height="537" alt="image" src="https://github.com/user-attachments/assets/c1bf6e22-8e17-4fb9-b1f9-a302434539a3" />
<img width="1113" height="541" alt="image" src="https://github.com/user-attachments/assets/dc3201d3-5a6c-461f-92d2-2a383836765f" />
<img width="1111" height="539" alt="image" src="https://github.com/user-attachments/assets/513aa566-68ec-4dec-b83b-147fd30b472d" />

rorychatt commented on frontend/src/widgets/charts/sharedUtils.ts:
```diff
@@ -258,7 +258,7 @@ export const generateYAxis = (
     },
     splitNumber: largeSpread ? 3 : 5,
     min: largeSpread ? safeTransform(minValue) : 0,
-    max: largeSpread ? safeTransform(maxValue) : 'dataMax',
```
good catch.

Maybe similar issue is with resizing?
